lib/internal/util/inspect.js

```diff
@@ -1753,7 +1781,14 @@ function formatArray(ctx, value, recurseTimes) {
     if (!ObjectPrototypeHasOwnProperty(value, i)) {
       return formatSpecialArray(ctx, value, recurseTimes, len, output, i);
     }
-    ArrayPrototypePush(output, formatProperty(ctx, value, recurseTimes, i, kArrayType));
+    const parentName = getParentName(ctx);
+    let formattedValue = formatProperty(ctx, value, recurseTimes, i, kArrayType);
+
+    if (ctx?.userOptions?.getSingularizeArrayValuesString) {
```

Comment:
This does not seem something we should do. It's a undocumented option that causes things to be formatted differently.

Reply:
do you have any suggestion about it? The only reason why I implemented this is because of the "Small quirk" section in the description 😄 

I am also not super found of it, if you have a better idea to solve the problem, I am all ears 😄